quaint/src/ast/values.rs

```diff
@@ -212,304 +672,316 @@ impl<'a> From<Value<'a>> for serde_json::Value {
             #[cfg(feature = "bigdecimal")]
-            Value::Numeric(r) => r.is_none(),
+            Self::Numeric(r) => r.is_none(),
             #[cfg(feature = "uuid")]
-            Value::Uuid(u) => u.is_none(),
-            Value::DateTime(dt) => dt.is_none(),
-            Value::Date(d) => d.is_none(),
-            Value::Time(t) => t.is_none(),
-            Value::Json(json) => json.is_none(),
+            Self::Uuid(u) => u.is_none(),
+            Self::DateTime(dt) => dt.is_none(),
+            Self::Date(d) => d.is_none(),
+            Self::Time(t) => t.is_none(),
+            Self::Json(json) => json.is_none(),
         }
     }
 
     /// `true` if the `Value` is text.
-    pub const fn is_text(&self) -> bool {
-        matches!(self, Value::Text(_))
+    pub(crate) fn is_text(&self) -> bool {
+        matches!(self, Self::Text(_))
     }
 
     /// Returns a &str if the value is text, otherwise `None`.
-    pub fn as_str(&self) -> Option<&str> {
+    pub(crate) fn as_str(&self) -> Option<&str> {
         match self {
-            Value::Text(Some(cow)) => Some(cow.borrow()),
-            Value::Bytes(Some(cow)) => std::str::from_utf8(cow.as_ref()).ok(),
+            Self::Text(Some(cow)) => Some(cow.borrow()),
+            Self::Bytes(Some(cow)) => std::str::from_utf8(cow.as_ref()).ok(),
             _ => None,
         }
     }
 
     /// Returns a char if the value is a char, otherwise `None`.
-    pub const fn as_char(&self) -> Option<char> {
+    pub(crate) fn as_char(&self) -> Option<char> {
         match self {
-            Value::Char(c) => *c,
+            Self::Char(c) => *c,
             _ => None,
         }
     }
 
     /// Returns a cloned String if the value is text, otherwise `None`.
-    pub fn to_string(&self) -> Option<String> {
+    pub(crate) fn to_string(&self) -> Option<String> {
         match self {
-            Value::Text(Some(cow)) => Some(cow.to_string()),
-            Value::Bytes(Some(cow)) => std::str::from_utf8(cow.as_ref()).map(|s| s.to_owned()).ok(),
+            Self::Text(Some(cow)) => Some(cow.to_string()),
+            Self::Bytes(Some(cow)) => std::str::from_utf8(cow.as_ref()).map(|s| s.to_owned()).ok(),
             _ => None,
         }
     }
 
     /// Transforms the `Value` to a `String` if it's text,
     /// otherwise `None`.
-    pub fn into_string(self) -> Option<String> {
+    pub(crate) fn into_string(self) -> Option<String> {
         match self {
-            Value::Text(Some(cow)) => Some(cow.into_owned()),
-            Value::Bytes(Some(cow)) => String::from_utf8(cow.into_owned()).ok(),
+            Self::Text(Some(cow)) => Some(cow.into_owned()),
+            Self::Bytes(Some(cow)) => String::from_utf8(cow.into_owned()).ok(),
             _ => None,
         }
     }
 
     /// Returns whether this value is the `Bytes` variant.
-    pub const fn is_bytes(&self) -> bool {
-        matches!(self, Value::Bytes(_))
+    pub(crate) fn is_bytes(&self) -> bool {
```

Comment:
Why do we lose the `const` here (and below)?

Reply:
All the methods, before guaranteed stack allocation. Those of the proxy don't anymore. We could make them const, but if later we use them in a constant scope, like a constant initialization, they will have to remain constant and their implementation won't be allowed to vary.  Given proxy methods in value cannot be const, I decided to dowgrade these as well as non-const.

Reply:
Ok then, thanks for providing the missing context.